src/org/sosy_lab/java_smt/solvers/cvc5/CVC5AbstractProver.java

```diff
@@ -243,6 +248,22 @@ public Optional<List<BooleanFormula>> unsatCoreOverAssumptions(
     throw new UnsupportedOperationException();
   }
 
+  @Override
+  public ProofNode getProof() {
+
+    Proof[] proofs = solver.getProof();
+    if (proofs == null || proofs.length == 0) {
+      throw new IllegalStateException("No proof available");
+    }
+
+    CVC5ProofProcessor pp = new CVC5ProofProcessor(creator);
+    try {
+      return pp.fromCVC5Proof(proofs[0]);
+    } catch (CVC5ApiException pE) {
+      throw new RuntimeException(pE);
```

Comment:
Re-throwing of exceptions should only happen for very good reasons, as they are much harder to work around than checked exceptions. A `ProverException` seems sensible here.